bot/exts/utilities/bookmark.py

```diff
@@ -123,15 +183,45 @@ async def _bookmark_context_menu_callback(self, interaction: discord.Interaction
     @commands.guild_only()
     @whitelist_override(roles=(Roles.everyone,))
     @commands.cooldown(1, 30, commands.BucketType.channel)
-    async def bookmark(self, ctx: commands.Context) -> None:
-        """Teach the invoker how to use the new context-menu based command for a smooth migration."""
-        await ctx.send(
-            embed=self.build_error_embed(
-                "The bookmark text command has been replaced with a context menu command!\n\n"
-                "To bookmark a message simply right-click (press and hold on mobile) "
-                "on a message, open the 'Apps' menu, and click 'Bookmark'."
+    async def bookmark(
+        self,
+        ctx: commands.Context,
+        target_message: Optional[WrappedMessageConverter],
+        *,
+        title: str = "Bookmark",
+    ) -> None:
+        """
+        Send the author a link to the specified message via DMs.
+
+        Members can either give a message as an argument, or reply to a message.
+
+        Bookmarks can subsequently be deleted by using the `bookmark delete` command in DMs.
+        """
+        target_message: Optional[discord.Message] = target_message or getattr(ctx.message.reference, "resolved", None)
+        if target_message is None:
+            raise commands.UserInputError(MESSAGE_NOT_FOUND_ERROR)
+
+        permissions = ctx.channel.permissions_for(ctx.author)
+        if not permissions.read_messages:
+            log.info(f"{ctx.author} tried to bookmark a message in #{ctx.channel} but has no permissions.")
+            embed = self.build_error_embed("You don't have permission to view this channel.")
+            await ctx.send(embed=embed)
+            return
+
+        try:
```

Comment:
So the idea should basically be:
```py
        view = SendBookmark(ctx.author, ctx.channel, target_message, title)
        try:
            await dm_bookmark(ctx.author, target_message, title)
        except discord.Forbidden:
            error_embed = self.build_error_embed(
                    f"{ctx.author.mention}, please enable your DMs to receive the bookmark."
            )
            await ctx.send(embed=error_embed)
        else:
            view.clicked.append(ctx.author.id)
            log.info(f"{ctx.author.mention} bookmarked {target_message.jump_url} with title '{title}'")
            
        embed = self.build_bookmark_embed(target_message)
        await ctx.send(embed=embed, view=view, delete_after=180)
```

This means that you change the ctor of `SendBookmark` to assign an empty list to `self.clicked` upon instantiation

Reply:
> This means that you change the ctor of SendBookmark to assign an empty list to self.clicked upon instantiation

I'm not sure I'm understanding what you mean by this, sorry

Reply:
A view is unique per interaction, and will live until it times out/bot dies.

What I'm saying is, the view is now instantiated with a list `click` that contains the id of the first person who invoked it.
You can't guarantee that the user is allowing DMs, so when that's not the case, the `clicked` list will contain that user's id whereas we couldn't really send the bookmark.

What I'm proposing to fix this is that you instantiate `clicked` to be an empty list, and you only append to it in the `else` section of your `try/except` block.